sdk/communication/communication-sms/test/smsClient.mocked.spec.ts

```diff
@@ -45,73 +77,153 @@ describe("[mocked] SmsClient", async () => {
     smsClient = new SmsClient(baseUri, new AzureKeyCredential("banana"), {
       httpClient: mockHttpClient
     });
+
     const spy = sinon.spy(mockHttpClient, "sendRequest");
-    const sendRequest: SendRequest = {
+    sinon.useFakeTimers();
```

Comment:
neat 👍

Reply:
Do we need to keep a reference to the returned clock and call `restore` on it?
We can also this and the restore into `beforeEach` and `afterEach` functions.